src/t_hash.c

```diff
@@ -922,3 +922,150 @@ void hscanCommand(client *c) {
         checkType(c,o,OBJ_HASH)) return;
     scanGenericCommand(c,o,cursor);
 }
+
+#define HRANDMEMBER_SUB_STRATEGY_MUL 3
+
+void hrandmemberWithCountCommand(client *c, long entryCount) {
+    unsigned long hashSize;
+    hashTypeIterator *hi;
+    robj *hash;
+    dict *d;
+    double randomDouble;
+    double threshold;
+    unsigned long index = 0;
+
+    if ((hash = lookupKeyReadOrReply(c,c->argv[1],shared.null[c->resp]))
+        == NULL || checkType(c,hash,OBJ_HASH)) return;
+
+    hashSize = hashTypeLength(hash);
+    if(entryCount > hashSize)
+        entryCount = hashSize;
+    addReplyMapLen(c, entryCount);
+    hi = hashTypeInitIterator(hash);
+
+    /* if the encoding is ziplist, to only iterate once, every time
+     * we try to pick a member, the probability we pick it is the 
+     * quotient of the count left we want to pick and the count still
+     * we haven't visited in the dict, this way, we could make every member
+     * be equally picked.*/
+    if(hash->encoding == OBJ_ENCODING_ZIPLIST) {
+        while (hashTypeNext(hi) != C_ERR && entryCount != 0) {
+            randomDouble = ((double)rand()) / RAND_MAX;
+            threshold = ((double)entryCount) / (hashSize - index);
+            if(randomDouble < threshold){
+                entryCount--;
+                addHashIteratorCursorToReply(c, hi, OBJ_HASH_KEY);
+                addHashIteratorCursorToReply(c, hi, OBJ_HASH_VALUE);
+            }
+            index++;
+        }
+    } else { 
+        /* if request count is more than we have, return all entries*/      
+        if(entryCount >= hashSize) {
+            while (hashTypeNext(hi) != C_ERR) {
+                addHashIteratorCursorToReply(c, hi, OBJ_HASH_KEY);
+                addHashIteratorCursorToReply(c, hi, OBJ_HASH_VALUE);
+            }
+            return;
+        }  
+
+        /* else we need a temporary dict to support our operations */
+        static dictType dt = {
+            dictSdsHash,                /* hash function */
+            NULL,                       /* key dup */
+            NULL,                       /* val dup */
+            dictSdsKeyCompare,          /* key compare */
+            NULL,                       /* key destructor */
+            NULL,                       /* val destructor */
+            NULL                        /* allow to expand */
+        };
+        d = dictCreate(&dt,NULL);    
```

Comment:
```suggestion
        d = dictCreate(&dt,NULL);
```
4 extra spaces at the end.